components/datetime/src/fields/symbols.rs

```diff
@@ -38,6 +38,92 @@ pub enum FieldSymbol {
     TimeZone(TimeZone),
 }
 
+impl FieldSymbol {
+    /// Symbols are necessary components of `Pattern` struct, which
+    /// uses efficient byte serialization and deserialization via `zerovec`.
+    ///
+    /// The `FieldSymbol` impl provides non-public methods that can be used to efficiently
+    /// convert between `u8` and the symbol variant.
+    ///
+    /// The serialization model packages the variant in one byte.
+    ///
+    /// 1) The top four bits are used to determine the type of the field.
+    ///    (Examples: `Year`, `Month`, `Hour`)
+    ///
+    /// 2) The bottom four bites are used to determine the symbol of the type.
+    ///    (Examples: `Year::Calendar`, `Hour::H11`)
```

Comment:
Isn't it the other way around?

```
        let k = kv & 0b0000_1111;
        let v = kv >> 4;
```

The lower four bits are `k`, and the upper four bits are `v`.  `0b0000_1111` has the lower four bits set, and `>> 4` drops the lower four bits and brings the upper four bits down.